data/publiccloud/terraform/gce.tf

```
@@ -90,10 +90,13 @@ resource "google_compute_instance" "openqa" {
         scopes = ["cloud-platform"]
     }
 
-    shielded_instance_config {
-        enable_secure_boot = "${var.uefi}"
-        enable_vtpm = "${var.uefi}"
-        enable_integrity_monitoring = "${var.uefi}"
+    dynamic "shielded_instance_config" {
+        for_each = var.uefi ? [random_id.service] : []
```

Nit comment, as we use this only as a flag, maybe it's not worth to make it clear in the file.
We could also define `type = bool` for that variable.
```suggestion
        for_each = var.uefi ? [ "NOT_USED" ] : []
        content {
            enable_secure_boot = true
            enable_vtpm = true
            enable_integrity_monitoring = true
        }
```